### DISCUSSION TITLE
Day / Time not displaying and link not working

### DISCUSSION THREAD

**Discussion by Studio88-KM** [code4recovery/12-step-meeting-list#834] [Q&A] [answered]
Hi there, love this plugin. Thank you for creating it!

We have a staging site here: https://wordpress-772911-2627160.cloudwaysapps.com/

Several issues after importing: 

1. We copied the Example CSV file and filled in client data. When we go to Meetings > Meetings, though, the Day and Time columns (the first two columns) are just showing "Appointment" instead of displaying the actual data whereas with the example CSV file, it was showing the days and times. I looked at both files to compare and I don't see anything different.
 
2. Also, when clicking on any link under "Meeting" or "Address", it does nothing even though it has data and we also added the MapBox token (and added the URL to the token). https://wordpress-772911-2627160.cloudwaysapps.com/index.php/meetings/. These links weren't working with the example data either. Don't think that's because it's a staging server but maybe? 

3. After importing the data, we got the message, "no location information provided for ___". Each of the entries it listed was for a Zoom meeting. But all the Zoom info was in the Conference URL and Conference URL Notes columns just like the other Zoom meetings that imported fine. 

What am I missing?

I did search the discussions for similar issues but couldn't find anything. Maybe I didn't find the right word or phrase. 

Thanks in advance!

**Comment by joshreisner** [comment, marked as answer]
hi! i think you've caught a bug on the inside page which we can fix. 

regarding "appointment" could you upload your CSV, or just a screenshot of the first few lines? I'm guessing something is slightly off with either the day or the time column.

**Reply by Studio88-KM** [reply]
Do you need access to the WP login or you're fixing the bug plugin wide?

Sure. [meetings-master-01.csv](https://github.com/code4recovery/12-step-meeting-list/files/8942530/meetings-master-01.csv)

Also, I just added a third issue I forgot to add first too. Could you help me with that?

**Reply by joshreisner** [reply]
Great, thanks! So regarding the appointment meetings, it looks correct but when I import it to my test site i have the same problem. I opened the sheet in Numbers, and re-exported to CSV and it works. Some kind of invisible formatting must have gotten messed up!
[meetings-master-02.csv](https://github.com/code4recovery/12-step-meeting-list/files/8942599/meetings-master-02.csv)

Regarding the inside page no need for access, it should start working for you soon.

Regarding the location info, we recommend adding general locations to online meetings. Eg `New York, NY` or `Brooklyn, NY`. 

Site looks great, congratulations!

**Reply by Studio88-KM** [reply]
Thanks so much! 

I have a PC so don't have Numbers. Do you have a suggestion for what to do from my end? Sorry, I may have missed what you're saying here. 

Or, if there's not an easy fix on a PC like there is with Mac, let me know if there are any suggestions about the invisible formatting. What to check, etc? We had the client do this so not quite sure what to look for since we used your example CSV. Maybe to enter the data in the top space instead of inside the cell or to if they're copying it from a doc to paste it into text editor first or.....?

![image](https://user-images.githubusercontent.com/77176347/174657795-ea716c10-6a4d-4494-a187-784b9ff89add.png)

Great, thanks about the last two things as well.

Thank you!

**Reply by joshreisner** [reply]
to my knowledge this issue is only with that CSV, so any program should work for exporting, eg MS Excel, or Google Sheets

once the data is all loaded, it probably would be best for the client to maintain it in wordpress rather than reimporting all the time

**Reply by Studio88-KM** [reply]
Okay, great. Thank you!

**Comment by Studio88-KM** [comment]
So the clicking on the links from the Meetings page is now working. Thanks for fixing that.

1. However, the Zoom information is not showing up on the single meeting display.  See example here: https://wordpress-772911-2627160.cloudwaysapps.com/index.php/meetings/?meeting=da-sought-through-prayer-meditation.  

1a. Also, the Zoom id and PW go into the Conference and Conference Notes columns on the spreadsheet but I don't see that when opening one of the meeting on the back end so where does it go on the backend? 

![image](https://user-images.githubusercontent.com/77176347/174832884-396bd17a-5fbd-4167-889e-5164277d7e4e.png)

1b. Shouldn't we change "Location" to Zoom instead of New York, NY on the spreadsheet so that people will see that's it's Zoom on the main meetings page under the Location column? 

2. How do we style the single meeting page? Do we need to do all our styling from the plugin file editor directly? It's looking wonky on the staging site we're working on: https://wordpress-772911-2627160.cloudwaysapps.com/index.php/meetings/?meeting=enough.

2a. Also curious why some single meeting entries/links show wider and some show really thin like this: 

![image](https://user-images.githubusercontent.com/77176347/174835890-341efc58-aca3-4c21-8c56-f88d4ad1e46f.png) 

Wider:

![image](https://user-images.githubusercontent.com/77176347/174836101-1b1ed738-8bb3-446e-b6e4-f0bfafe45b39.png)

4. About the Meetings display list, for some reason, ours is starting with Tuesday and then at the very bottom there's a single Tuesday after the Mondays that's the only AM meeting. And, it looks like there's a 7 AM Tuesday meeting missing. 

6. I see a bunch of meetings missing from the import actually. Hmmm?

7. Can we set the meeting list display to start on the current day like Google calendar does? And if not, can we start on Monday or even Sunday? 

8. On the backend Meetings > Search, the search doesn't work when searching for a day/time, etc only region? If so, can that be a feature request to be able to search for meetings by any field? 

![image](https://user-images.githubusercontent.com/77176347/174833016-f28e84f8-54eb-4b69-b29c-2df6321c4af4.png)

9. Is there a way to add field for yes/no to indicate if it's a hybrid meeting on the spreadsheet so we/client won't have to go through each meeting and fill certain things in manually? Or, is that triggered by adding an address?

10. Also, curious if there's something we need to add to each Meeting card/backend to get the Add to Calendar to include the call in instructions, Zoom instruction, address, etc? Right now, it just adds a blank card with date/time. If not, I'd love that as a feature request at some point. I think that would be really helpful for the user. 

![image](https://user-images.githubusercontent.com/77176347/174834395-ab32285e-94dd-4063-8b5b-7c9ca759fcdd.png)

**Reply by joshreisner** [reply]
that's a lot of stuff 😅 

1. i dont see this meeting on the CSV, looking at the CSV it looks like these Conference URLs were not entered properly. they need to be full zoom join links, eg `https://zoom.us/j/123456789` - if you fix that I think this will take care of most of the issues. We recommend the location name "Online Meeting" and the Address "New York, NY, USA" for these meetings.
2. we have two appearances for our meeting finder, TSML UI and Legacy UI. It may be that you like Legacy UI better since it starts at the beginning of the current day (TSML UI begins at now). if you want to stick with TSML UI then i would add some custom CSS because your theme is not giving the meeting finder enough room. eg `#content .wp-site-blocks { width: 100%; }`
3. (looks like you skipped 3)
4. see answer 2. you can change appearances at import & settings > settings tab > user interface
5. yeah in your original CSV i only see mon-wed meetings
6. see answer 2
7. i think it also searches meeting name and notes on the backend. we [do have a feature request issue](https://github.com/code4recovery/12-step-meeting-list/issues/370) to expand this search
8. (looks like you skipped 8)
9. no, but i think once the online meetings are in properly you will see them show up as hybrid
10. that "add to calendar" is a TSML UI-only feature. once all the info is in properly, if you're still using TSML UI and still seeing an issue here, please create a ticket: https://github.com/code4recovery/tsml-ui/issues you can see how it work on another site here: https://aasanjose.org/meetings

**Reply by Studio88-KM** [reply]
I know. Thanks so much for answering them! I wasn't sure if I should start another string. 

1. Great on the Zoom issue. I'll ask client to change to that URL format. 

2. Oh! I didn't realize it was starting now. Duh! That's perfect.

3. Woops.

4. Thanks!

5. I uploaded an updated file with about 80 or more meetings. There are meetings missing.

6. Thanks.

7. Thanks!

8. Weird. The system was generating the numbers so next time I'll make sure it does it correctly.

9. Okay, thanks. 

10. Okay, will do. Thanks!

**Reply by joshreisner** [reply]
meetings might not appear if the location isn't valid. you might check a few of the meetings that arent being loaded and see if that's the case?

**Reply by Studio88-KM** [reply]
I check the AA site and tried clicking on Add to Calendar but it didn't show the address, etc. 

![image](https://user-images.githubusercontent.com/77176347/174889584-9a01bee6-2f98-409a-8082-8f6d08d2ec22.png)

I tried several meetings.

**Reply by joshreisner** [reply]
ok https://github.com/code4recovery/tsml-ui/issues/234
